source/slang/hlsl.meta.slang

```diff
@@ -9636,12 +10548,16 @@ void sincos(T x, out T s, out T c)
 
 __generic<T : __BuiltinFloatingPointType, let N : int>
 [__readNone]
-[require(cpp_cuda_glsl_hlsl_spirv, sm_2_0_GLSL_140)]
+[ForceInline]
+[require(cpp_cuda_glsl_hlsl_metal_spirv, sm_2_0_GLSL_140)]
 void sincos(vector<T,N> x, out vector<T,N> s, out vector<T,N> c)
 {
     __target_switch
     {
     case hlsl: __intrinsic_asm "sincos";
+    case metal:
+        s = __sincos_metal(x, c);
```

Comment:
Can be 
*($1)=sincos($0,*($2))

Reply:
Oh! I will give a try.

Reply:
This should work but it is currently not working.
The error message is
```
metal 32023.155: diff.meta.slang(1979): error :  unknown type name 'out'
void _sincos_impl_0(float x_0, out float s_0, out float c_0)
```
It sounds like we cannot use `out` in Metal.
I guess we should emit `&` istead of `out`.

Reply:
Yeah it needs to be 
T thread&

All pointers and references need an address space. We can use thread now but we will need a pass that derives and specializes functions with address spaces.